.github/CONTRIBUTING.md

```diff
@@ -28,40 +29,87 @@ To sign the Eclipse CLA you need to:
+For example: `issues/#276-rdfxml-sax` is the branch for a fix for
+GitHub issue #276, which has to do with RDF/XML and the SAX parser.
 
-If you're unsure on which branch a contribution should be made, please ask!
+Once a feature/fix is complete, tested, and reviewed (via a Pull Request), it
+is merged into master, and the issue branch is closed.
 
-### A note on the 1.0.x release branch and Java 7 compatibility ###
+### Releases
 
-RDF4J 1.0 is a **backport** of RDF4J 2.0, to be executable on a Java 7 Runtime Environment. This means that we do not accept any fixes or feature development directly against the `releases/1.0.x` branch, and we never merge this branch back into either `master`, or the `releases/2.0.x` branch. 
+RDF4J is on an 8-week release cycle for minor and major releases. See the [Milestone overview](https://github.com/eclipse/rdf4j/milestones) in the [issue tracker](https://github.com/eclipse/rdf4j/issues) for an overview of planned releases.
 
-Instead, new features are developed against the `master` branch, and these are never merged into either release branch (they are instead scheduled for inclusion in the next minor or major release: 2.1, or 3.0). Bug fixes and minor improvements (which are sufficiently minor that they can be considered part of a patch release) are developed against the `releases/2.0.x` branch, which can then be merged back into `releases/1.0.x` branch. 
+In order to guarantee stability of the release and to ensure we have enough
+time to complete release review, we use a feature cutoff date. This date is
+typically three weeks before the release date. The feature cutoff date does not
+mean that each feature needs to be complete, but it does mean that we ask every
+contributor to make a commitment to have their feature complete and stable in
+time for final review and release. If a contributor cannot make that
+commitment, the feature is not included in the release and is simply
+"postponed" to the next release cycle.
 
-Since `releases/1.0.x` is to be kept compatible with Java 7, it is inevitable that it will occassionally become unstable when merging the `releases/2.0.x` branch into it: the build process verifies that the branch is Java 7-compatible, so it will fail with a compilation error when this happens. These compilation failures can be fixed by branching a hotfix branch directly directly off the `releases/1.0.x` branch.
+We occassionally also schedule *patch releases*, which only include (backward
+compatible) bug fixes. These releases are typically planned on short notice and
+can be done without much in the way of formal review.
+
+#### Release branches
+
+A few days before release of a minor or major version, we create a *release
+branch* from the current head of the master branch. The purpose of this release
+branch is to perform last-minute tweaks and little fixes. Once we are ready to
+release, the head of the release branch is tagged with the version number, and
```

Comment:
Replacing "head", which is technically correct, with "latest commit" may make this slightly more fluent.